OMCompiler/Compiler/NFFrontEnd/NFConnectEquations.mo

```diff
@@ -222,8 +222,13 @@ algorithm
       end for;
     end if;
   else
-    equations := list(makeEqualityAssert(c1.name, c1.source, c2.name, c2.source)
-      for c2 in listRest(elements));
+    // don't create asserts for empty arrays
+    if Type.isEmptyArray(ComponentRef.getSubscriptedType(c1.name)) then
```

Comment:
Connectors store their type, so you can use `c1.ty` instead of `ComponentRef.getSubscriptedType(c1.name)`.